lib/app/helpers/user.ex

```diff
@@ -0,0 +1,17 @@
+defmodule App.Helpers.UserHelper do
+  @moduledoc """
+  Helpers function for users
+  """
+  alias App.{Repo, User}
+
+  def udpate_or_update_user(params) do
+    user = Repo.get_by(User, user_id: params.user_id)
+    if user do
+      changeset = User.changeset(user, params)
+      Repo.update!(changeset)
+    else
+      changeset = User.changeset(%User{}, params)
+      Repo.insert!(changeset)
+    end
+  end
```

Comment:
Following this example: https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_or_update/2-example and making use of [`insert_or_update!`](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_or_update!/2)

```elixir
def update_or_update_user(%{user_id: user_id} = params) do
  case Repo.get_by(User, user_id: user_id) do
    nil -> %User{user_id: user_id}
    user -> user
  end
  |> User.changeset(params)
  |> Repo.insert_or_update!
end
```
I also think calling this function `insert_or_update_user` fits with `Repo`'s naming better

Reply:
:+1:  indeed the name can be updated